### PR TITLE
Pfappserver event module

### PR DESCRIPTION
**Pull request by satkunas**
# Description
A global singleton store module that binds on window and document events. Reduces the work needed for a template to bind/unbind and track events. Improves performance since bind/unbind is no longer used elsewhere.

# Impacts
Replaces the `[escape]` and `[ctrl]` key binds. 

# Delete branch after merge
YES

# Checklist
- [n/a] Document the feature
- [n/a] Add unit tests
- [n/a] Add acceptance tests (TestLink)

## Enhancements
* centralized event bindings
* add hotkeys to main navigation and section filter

### CODE REVIEW

**Comment by satkunas** [comment]
@cgx please test to ensure that commit https://github.com/inverse-inc/packetfence/pull/4567/commits/734cf6b541ebe2dcf44f38461a97181532fcb2da works as expected in security event triggers.

File: html/pfappserver/root/static.alt/src/store/modules/events.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  altShiftAKey: state => state.focus && state.documentEvent.altKey && state.documentEvent.shiftKey && state.documentEvent.keyCode === state.keyCodes['KeyA'],
+  altShiftCKey: state => state.focus && state.documentEvent.altKey && state.documentEvent.shiftKey && state.documentEvent.keyCode === state.keyCodes['KeyC'],
+  altShiftFKey: state => state.focus && state.documentEvent.altKey && state.documentEvent.shiftKey && state.documentEvent.keyCode === state.keyCodes['KeyF'],
+  altShiftNKey: state => state.focus && state.documentEvent.altKey && state.documentEvent.shiftKey && state.documentEvent.keyCode === state.keyCodes['KeyN'],
+  altShiftRKey: state => state.focus && state.documentEvent.altKey && state.documentEvent.shiftKey && state.documentEvent.keyCode === state.keyCodes['KeyR'],
+  altShiftSKey: state => state.focus && state.documentEvent.altKey && state.documentEvent.shiftKey && state.documentEvent.keyCode === state.keyCodes['KeyS'],
+  altShiftUKey: state => state.focus && state.documentEvent.altKey && state.documentEvent.shiftKey && state.documentEvent.keyCode === state.keyCodes['KeyU'],
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/store/modules/events.js]: Remove trailing comma.

File: html/pfappserver/root/static.alt/src/store/modules/events.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+const actions = {
+  bind: ({ commit, dispatch }) => {
+    document.body.addEventListener('mousedown', (event) => dispatch('onBodyMouseDown', event))
+    document.body.addEventListener('mouseup', (event) => dispatch('onBodyMouseUp', event))
+    document.addEventListener('keydown', (event) => dispatch('onKeyDown', event))
+    document.addEventListener('keyup', (event) => dispatch('onKeyUp', event))
+    window.addEventListener('blur', (event) => dispatch('onBlur', event))
+    window.addEventListener('focus', (event) => dispatch('onFocus', event))
+    window.addEventListener('resize', (event) => dispatch('onResize', event))
+    commit('RESIZE', null)  // init windowSize
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/store/modules/events.js]: Remove extra space before comment delimiter.

File: html/pfappserver/root/static.alt/src/store/modules/events.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    Slash: 191,
+    Backquote: 192,
+    BracketLeft: 219,
+    Backslash: 220,
+    BracketRight: 221,
+    Quote: 222
+  },
+  keyDown: false,
+  mouseDown: false,
+  focus: true,
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/store/modules/events.js]: Why not initializing `focus` to `false`?

File: html/pfappserver/root/static.alt/src/store/modules/events.js [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  isKey: state => key => (state.focus && (key in state.keyCodes) && (state.documentEvent.keyCode || null) === state.keyCodes[key]),
+  isKeyCode: state => keyCode => (state.focus && (state.documentEvent.keyCode || null) === keyCode),
+  keyCode: state => state.documentEvent.keyCode,
+  focus: state => state.focus,
+  keyDown: state => state.focus && state.keyDown,
+  mouseDown: state => state.focus && state.mouseDown,
+  altKey: state => state.focus && state.documentEvent.altKey,
+  ctrlKey: state => state.focus && state.documentEvent.ctrlKey,
+  shiftKey: state => state.focus && state.documentEvent.shiftKey,
+  escapeKey: state => state.focus && state.documentEvent.keyCode === state.keyCodes['Escape'],
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/store/modules/events.js]: When using constant keys, use the syntax `state.keyCodes.Escape`.

**Review by cgx — CHANGES_REQUESTED** [review]
Could you also cleanup `NodesImport.vue` and `UsersImport.vue` where we have keydown listeners? Remove the references to `this.noInitBindKeys` which is not defined for those components.

**Review by cgx — APPROVED** [review]

**Pull request merged**